solidity/random-beacon/contracts/libraries/Groups.sol

```diff
@@ -69,28 +70,33 @@ library Groups {
 
     /// @notice Sets addresses of members for the group eliminating members at
     ///         positions pointed by the misbehaved array.
+    ///
+    ///         NOTE THAT THIS FUNCTION CHANGES ORDER OF MEMBERS IN THE GROUP
+    ///         IF THERE IS AT LEAST ONE MISBEHAVED MEMBER
+    ///
+    ///         The final group members indexes should be obtained post-DKG
+    ///         and they may differ from the ones outputted by the group
+    ///         selection protocol.
     /// @param group The group storage.
     /// @param members Group member addresses as outputted by the group selection
     ///        protocol.
-    /// @param misbehaved Array of misbehaved (disqualified or inactive) group
-    ///        members indices in ascending order; Indices reflect positions
+    /// @param misbehavedMembersIndices Array of misbehaved (disqualified or
+    ///        inactive) group members. Indices reflect positions
     ///        of members in the group as outputted by the group selection
-    ///        protocol - member indices start from 1.
-    // TODO: This function should be optimized for members storing.
-    // See https://github.com/keep-network/keep-core/pull/2666/files#r732629138
+    ///        protocol.
     function setGroupMembers(
         Group storage group,
         uint32[] calldata members,
-        uint8[] calldata misbehaved
+        uint8[] calldata misbehavedMembersIndices
     ) private {
         group.members = members;
 
         // Iterate misbehaved array backwards, replace misbehaved
         // member with the last element and reduce array length
-        uint256 i = misbehaved.length;
+        uint256 i = misbehavedMembersIndices.length;
         while (i > 0) {
             // group member indices start from 1, so we need to -1 on misbehaved
```

Comment:
There are a couple of `misbehaved` words left in the comments in this contract. Thoughts on replacing them with `misbehavedMembersIndices` as well?

Reply:
Please see https://github.com/keep-network/keep-core/pull/2679/commits/ab7ee6c947d7fc67f7c881aa1944d1330275718e. I changed just one, I think other occurrences are better the way they are. E.g. "iterate misbehaved array backwards" in the comment to the code is easier to read and follow than "iterate misbehavedMembersIndices array backwards". :wink: